xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-api/src/main/java/org/xwiki/security/authentication/api/ResetPasswordManager.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.security.authentication.api;
+
+import javax.mail.internet.InternetAddress;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+import org.xwiki.user.UserReference;
+
+/**
+ * Component dedicated to handle the reset password operation.
+ * This component is designed to handle a reset password in 3 steps:
+ *   1. a request is performed for doing a reset password for a given user: a verification code is transmitted to the
+ *      user by a side way (e.g. email)
+ *   2. the user certifies her identity by sending back the verification code
+ *   3. the user specify a new password which is updated internally.
+ *
+ * @version $Id$
+ * @since 13.1RC1
+ */
+@Role
+@Unstable
+public interface ResetPasswordManager
+{
+    /**
+     * Dedicated response interface for the first reset password request.
+     *
+     * @version $Id$
+     * @since 13.1RC1
+     */
+    @Unstable
+    interface ResetPasswordRequestResponse
+    {
+        /**
+         * @return the reference of the user for whom the reset password request have been performed.
+         */
+        UserReference getUserReference();
+
+        /**
+         * @return the email address of the user for whom the reset password request have been performed.
+         */
+        InternetAddress getUserEmail();
```

Comment:
Do we need this, knowing that the UserReference allows getting the email of the user? I don't know enough to judge, just asking in case i's useful.

Reply:
That mainly avoids having to compute back the email address from the user reference in the script service when it has always been computed in the main component.